contracts/ETO/ETOCommitment.sol

```diff
@@ -916,58 +927,57 @@ contract ETOCommitment is
         // write in single pack, hopefully storage will be optimized...
         ticket.equivEurUlps = uint96(newEquivEurUlps);
         ticket.rewardNmkUlps = newRewardNmkUlps;
-        ticket.equityTokenInt = newEquityTokenInt;
+        ticket.equityTokenAmount = newEquityTokenAmount;
         ticket.amountEth = newAmountEth;
         ticket.amountEurUlps = newAmountEurUlps;
         ticket.usedLockedAccount = usedLockedAccount;
     }
 
     function updateTotalInvestment(
-        uint256 equityTokenInt,
-        uint256 fixedSlotEquityTokenInt,
+        uint256 equityTokenAmount,
+        uint256 fixedSlotEquityTokenAmount,
         uint256 equivEurUlps,
         uint256 amount,
         bool firstTimeInvestment
     )
         private
     {
-        // total number of tokens must fit into uint56, uints are coerced to biggest size
-        uint256 newTotalTokensInt = _totalTokensInt + equityTokenInt;
-        assert(newTotalTokensInt < 2**56);
-        // as fixedSlotEquityTokenInt256 always lte equityTokenInt no chances to overflow
-        uint56 newTotalFixedSlotsTokensInt = _totalFixedSlotsTokensInt + uint56(fixedSlotEquityTokenInt);
+        // total number of tokens must fit into uint128, uints are coerced to biggest size
+        // as ticket token storage is 2**96, we need 2**32 tickets of max size to overflow so disregard
+        uint128 newTotalTokenAmount = uint128(_totalTokenAmount + equityTokenAmount);
+        // as fixedSlotEquityTokenAmount always lte equityTokenAmount no chances to overflow
+        uint128 newTotalFixedSlotsTokenAmount = _totalFixedSlotsTokenAmount + uint128(fixedSlotEquityTokenAmount);
         // add new investor
-        uint32 newTotalInvestors = _totalInvestors + (firstTimeInvestment ? 1 : 0);
-        // new total eur equivalent invested
-        uint256 newTotalEquivEurUlps = _totalEquivEurUlps + equivEurUlps;
-        assert(newTotalEquivEurUlps < 2**112);
-        // resue eth/neur contribution slots for native currency bookkeeping
-        uint112 newTotalEth = _additionalContributionEth;
-        uint112 newTotalEurUlps = _additionalContributionEurUlps;
+        uint128 newTotalInvestors = _totalInvestors + (firstTimeInvestment ? 1 : 0);
+        // new total eur equivalent invested, disregard overflow for the same reason as with tokens
+        uint128 newTotalEquivEurUlps = uint128(_totalEquivEurUlps + equivEurUlps);
```

Comment:
uint128 -> uint256 intended?

>  new total eur equivalent invested, disregard overflow for the same reason as with tokens
let's talk live about this!

Reply:
let's go through the code of `updateTotalInvestment` again. there are indeed some "practical" assumptions like there will be less than 2^32 investments in single eto. we can add checked math/comments if things are not clear

we can add checked math everywhere, I prefer to track all arithmetic to be sure if it can overflow or not. asserting overflows can put you in dead end anyway...